anesthetic/samples.py

```diff
@@ -447,7 +447,8 @@ def beta(self):
     @beta.setter
     def beta(self, beta):
         self._beta = beta
-        logw = self.dlogX() + self.beta*self.logL
+        logw = self.dlogX() + numpy.where(self.logL == -numpy.inf,
+                                          -numpy.inf, self.beta * self.logL)
         self._weight = numpy.exp(logw - logw.max())
```

Comment:
Note that the modified line calculates `logw` not `w`, which is why `np.where` needs to return `-np.inf` instead of `0`.